src/power_grid_model/validation/rules.py

```diff
@@ -697,6 +698,36 @@ def none_missing(data: SingleDataset, component: str, fields: Union[str, List[st
     return errors
 
 
+def valid_p_q_sigma(data: SingleDataset, component: str) -> List[MultiFieldValidationError]:
+    """
+    Check validity of the pair `(p_sigma, q_sigma)` for 'sym_power_sensor' and 'asym_power_sensor'.
+
+    Args:
+        data: The input/update data set for all components
+        component: The component of interest, in this case only 'sym_power_sensor' or 'asym_power_sensor'
+
+    Returns:
+        A list containing zero or one MultiFieldValidationError, listing the p_sigma and q_sigma mismatch.
+        Note that with asymetric power sensors, partial assignment of p_sigma and q_sigma is also considered mismatch.
+    """
+    errors = []
+    p_sigma = data[component]["p_sigma"]
+    q_sigma = data[component]["q_sigma"]
+    ids = [data[component]["id"].flatten().tolist()]
+    p_nan = np.isnan(p_sigma)
+    q_nan = np.isnan(q_sigma)
+    p_inf = np.isinf(p_sigma)
+    q_inf = np.isinf(q_sigma)
+    mis_match = p_nan != q_nan
+    mis_match |= np.logical_xor(p_nan, q_nan).any()
+    if component == "asym_power_sensor":
+        mis_match |= p_nan.any() == (~p_nan).any()
+        mis_match |= q_nan.any() == (~q_nan).any()
+    if mis_match.any() or np.logical_or(p_inf.any(), q_inf.any()):
+        errors.append(MultiFieldValidationError(component, ["p_sigma", "q_sigma"], ids))
```

Comment:
you need to replace all instances of `.any()` with `.any(axis=-1)` for `"asym_power_sensor"`, and remove it entirely for `sym_power_sensor`.

`.any()` causes the entire list across all id's to be squashed into a single boolean value. We only want to return the ids that were incorrect. `mis_match` should therefore be an array of shape `data[component].shape` (note that for asymmetric sensors, `data[component]` still has the same shape, but `data[component]['p_sigma']` and `data[component]['q_sigma']` again are numpy arrays but with an additional dimension.

The following is just for illustration. The input is a `SingleDataset`, not a `BatchDataset`, so the first dimension drops, and just `axis=1` should do the trick as well, but `axis=-1` is more clear in this case

```py
>>> import power_grid_model as pgm
>>> asym_ps = pgm.initialize_array("input", "asym_power_sensor", (3, 2))
>>> asym_ps.shape
(3, 2)
>>> asym_ps["p_sigma"].shape
(3, 2, 3)
>>> asym_ps["q_sigma"].shape
(3, 2, 3)
```

`np.logical_xor` makes an element-wise comparison between two arrays of the same shape:

```py
>>> import numpy as np
>>> np.logical_xor(asym_ps["p_sigma"], asym_ps["q_sigma"]).shape
(3, 2, 3)
>>> np.logical_xor(asym_ps["p_sigma"], asym_ps["q_sigma"]).any()
False
```

Instead, the `mis_match` should become an array of the same shape (as you already do earlier):

```py
>>> mismatch = asym_ps["p_sigma"] != asym_ps["q_sigma"]
>>> mismatch.shape
(3, 2, 3)
```

I think the complete logic should be something like this (taking inspiration from and following the conventions of the function in the above: https://github.com/PowerGridModel/power-grid-model/pull/425/files#diff-216fb00ddcbaef2c504f2b9244c35a587fe21ec8b3fda59130ec57d58e49d3e5R688-R697 )

```suggestion
    p_nan = np.isnan(p_sigma)
    q_nan = np.isnan(q_sigma)
    p_inf = np.isinf(p_sigma)
    q_inf = np.isinf(q_sigma)
    if p_sigma.ndim > 1:
        p_sigma = p_sigma.any(axis=-1)
        q_sigma = p_sigma.any(axis=-1)
        p_inf = p_inf.any(axis=-1)
        q_inf = q_inf.any(axis=-1)

    invalid = (p_nan != q_nan) or p_inf or q_inf
    if invalid.any():
        ids = data[component]["id"][invalid].flatten().tolist()
        errors.append(MultiFieldValueError(component, ["p_sigma", "q_sigma"], ids))
```

Reply:
why did you `unresolve` this? i don't see any changes yet

Reply:
It's on my local files, changes not yet pushed...

Reply:
I updated `invalid = (p_nan != q_nan) or p_inf or q_inf`, as it causes issues with ambiguity. Rest incoming.